cmd/thanos/query.go

```diff
@@ -733,20 +737,29 @@ func removeDuplicateEndpointSpecs(logger log.Logger, duplicatedStores prometheus
 	return deduplicated
 }
 
-// firstDuplicate returns the first duplicate string in the given string slice
-// or empty string if none was found.
-func firstDuplicate(ss []string) string {
+// validateAddrs checks an address slice for duplicates and empty or invalid elements.
+func validateAddrs(addrs []string) error {
 	set := map[string]struct{}{}
 
-	for _, s := range ss {
-		if _, ok := set[s]; ok {
-			return s
+	for _, addr := range addrs {
+		if addr == "" {
+			return errors.New("Address is empty.")
+		}
+
+		qtypeAndName := strings.SplitN(addr, "+", 2)
+		hostAndPort := strings.SplitN(addr, ":", 2)
+		if len(qtypeAndName) != 2 && len(hostAndPort) != 2 {
+			return errors.Errorf("Address %s is not of <host>:<port> format or a valid DNS query.", addr)
 		}
 
-		set[s] = struct{}{}
```

Comment:
Maybe we can also call https://pkg.go.dev/net#ParseIP here to check `hostAndPort[0]`?